privacyidea/lib/utils/__init__.py

```diff
@@ -58,6 +58,8 @@
 
 ALLOWED_SERIAL = "^[0-9a-zA-Z\-_]+$"
 
+SPECIAL_CHARS_LIST = string.punctuation
+SPECIAL_CHARS_REGXEP = r"[" + SPECIAL_CHARS_LIST + "]"
```

Comment:
one reason for the failing tests was a mismatch in generation and checking of the pin.

Reply:
The SPECICAL_CHARS_LIST is not the same anymore, E.g. "§" is missing.

Also I think SPECIAL_CHARS_REGEXP is now wrong. The brackets [ and ] need to be escaped. Otherwise they will work as a group.

Reply:
okay, I reverted it. However, I removed the square brackets, since they seem not to behave the same for string and regexp. We may discuss this.
It is crucially important that SPECIAL_CHARS_LIST and SPECIAL_CHARS_REGXEP are the same, since one is for generation and one for checking of the generated pins.

Reply:
because of this, test test_20_pin_policy atm fails at https://github.com/privacyidea/privacyidea/blob/2446d9a6eaf0f68bff483791912802a0018c463e/tests/test_lib_utils.py#L520

Reply:
added a base string identifier r"[[[". Now those work again. Python interprets two square brackets as one string and the third one is left, causing trouble.